deepchecks/checks/performance/model_error_analysis.py

```diff
@@ -0,0 +1,317 @@
+#
+"""Module of segment performance check."""
+from typing import Callable, Dict, Tuple
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+from category_encoders import TargetEncoder
+from sklearn.compose import ColumnTransformer
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.impute import SimpleImputer
+from sklearn.metrics import r2_score
+from sklearn.pipeline import Pipeline
+from sklearn.tree import DecisionTreeRegressor
+
+from deepchecks import Dataset, CheckResult, TrainTestBaseCheck, ConditionResult, ConditionCategory
+from deepchecks.errors import DeepchecksProcessError
+from deepchecks.utils.features import calculate_feature_importance
+from deepchecks.utils.metrics import task_type_check, ModelType, initialize_multi_scorers, get_scorers_list, \
+    get_scorer_single
+from deepchecks.utils.plot import colors
+from deepchecks.utils.strings import format_number, format_percent
+from deepchecks.utils.validation import validate_model
+
+__all__ = ['ModelErrorAnalysis']
+
+
+class ModelErrorAnalysis(TrainTestBaseCheck):
+    """Find features that best split the data into segments of high and low model error.
+
+    The check trains a regression model to predict the error of the user's model. Then, the features scoring the highest
+    feature importance for the error regression model are selected and the distribution of the error vs the feature
+    values is plotted. The check results are shown only if the error regression model manages to predict the error
+    well enough.
+
+    Args:
+        max_features (int): maximal number of features to show. (default: 3)
+        min_feature_contribution (float): minimum feature importance of a feature to the error regression model
+            in order to show the feature. (default: 0.15)
+        min_error_model_score (float): minimum r^2 score of the error regression model for displaying the
+            check. (default: 0.5)
+        min_segment_size (float): minimal fraction of data that can comprise a weak segment. (default: 0.1)
+        alternative_scorer (Dict[str, Callable], default None):
+            An optional dictionary of scorer name to scorer function. Only a single entry is allowed in this check.
+            If none given, using default scorer
+        n_samples (int): number of samples to use for this check. (default: 50000)
+        n_display_samples (int): number of samples to display in scatter plot. (default: 5000)
+        random_seed (int): random seed for all check internals. (default: 42)
+    """
+
+    def __init__(
+            self,
+            max_features: int = 3,
+            min_feature_contribution: float = 0.15,
+            min_error_model_score: float = 0.5,
+            min_segment_size: float = 0.05,
+            alternative_scorer: Dict[str, Callable] = None,
+            n_samples: int = 50_000,
+            n_display_samples: int = 5_000,
```

Comment:
well these aren't numbers

Reply:
It's legit in python, as the great @matanper taught me

Reply:
Cool.